bolt/src/main/java/com/slack/api/bolt/service/builtin/AmazonS3OAuthStateService.java

```diff
@@ -33,74 +33,91 @@ public AmazonS3OAuthStateService(String bucketName) {
     public Initializer initializer() {
         return (app) -> {
             // The first access to S3 tends to be slow on AWS Lambda.
-            AWSCredentials credentials = getCredentials();
-            if (credentials == null || credentials.getAWSAccessKeyId() == null) {
+            this.credentialsProvider = DefaultCredentialsProvider.create();
+            AwsCredentials credentials = createCredentials(credentialsProvider);
+            if (credentials == null || credentials.accessKeyId() == null) {
                 throw new IllegalStateException("AWS credentials not found");
             }
             if (log.isDebugEnabled()) {
-                log.debug("AWS credentials loaded (access key id: {})", credentials.getAWSAccessKeyId());
+                log.debug("AWS credentials loaded (access key id: {})", credentials.accessKeyId());
+            }
+            boolean bucketExists = false;
+            Exception error = null;
+            try (S3Client s3 = createS3Client()) {
+                bucketExists = s3.headBucket(HeadBucketRequest.builder().bucket(bucketName).build()) != null;
```

Comment:
@seratch San: Do you think that [this](https://docs.aws.amazon.com/ja_jp/AmazonS3/latest/API/s3_example_s3_Scenario_DoesBucketExist_section.html) implementation from the AWS is better than then using ``HeadBucketRequest``

Reply:
@acharyashashank Thanks for the suggestion! 

I switched to headBucket because the AWS migration guide recommends using the API in version 2: https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/migration-service-changes.html. Do you think using getBucketAcl instead could bring benefits such as faster runtime performance?

Even if I use the code snippet provided there, I won't follow AwsServiceException error handling because the goal of this code is to verify if the client has permissions to fetch/modify S3 bucket data, rather than checking the bucket's existence.